yorkie/src/main/kotlin/dev/yorkie/document/crdt/ElementRht.kt

```diff
@@ -0,0 +1,156 @@
+package dev.yorkie.document.crdt
+
+import dev.yorkie.document.time.TimeTicket
+import dev.yorkie.util.YorkieLogger
+
+/**
+ * [ElementRht] is a hashtable with logical clock(Replicated hashtable).
+ */
+internal class ElementRht<T : CrdtElement> : Iterable<ElementRht.Node<T>> {
+    private val logTag = "ElementRhtTag"
+
+    private val nodeMapByKey: MutableMap<String, Node<T>> = mutableMapOf()
+
+    private val nodeMapByCreatedAt: MutableMap<TimeTicket, Node<T>> = mutableMapOf()
+
+    /**
+     * Sets the [value] using the given [key].
+     * If the object exists in [nodeMapByKey] by same [key] then return [CrdtElement], otherwise null.
+     */
+    operator fun set(key: String, value: T): T? {
+        var removed: T? = null
+        val node = nodeMapByKey[key]
+        if (node != null && !node.isRemoved && node.remove(value.createdAt)) {
+            removed = node.value
+        }
+
+        val newNode = Node(key, value)
+        nodeMapByCreatedAt[value.createdAt] = newNode
+        if (node == null || node.value.createdAt < value.createdAt) {
+            nodeMapByKey[key] = newNode
+        }
+        return removed
+    }
+
+    /**
+     * Removes the element in [nodeMapByCreatedAt] using the given key [createdAt],
+     * and [TimeTicket] will be removed by [executedAt].
+     * If the object exists in [nodeMapByCreatedAt] by same key [createdAt], it will return.
+     *
+     * @throws NoSuchElementException if [Node] doesn't exist.
+     */
+    fun remove(createdAt: TimeTicket, executedAt: TimeTicket): T {
+        val node = nodeMapByCreatedAt[createdAt]
+            ?: throw NoSuchElementException("The ElementRhtNode by $createdAt doesn't exist.")
+        node.remove(executedAt)
+        return node.value
+    }
+
+    /**
+     * Removes the element in [nodeMapByKey] using the given [key] and [removedAt].
+     * If the element removed successfully, removed [CrdtElement] will return.
+     *
+     * @throws NoSuchElementException if [Node] doesn't exist.
+     */
+    fun removeByKey(key: String, removedAt: TimeTicket): T {
+        val node = nodeMapByKey[key]
+            ?: throw NoSuchElementException("$key doesn't exist in nodeMapByKey")
+        node.remove(removedAt)
+        return node.value
```

Comment:
@hackerwins I returned the `node.value` in the same way as the `remove()` function above. This approach differs slightly from the [JS-SDK](https://github.com/yorkie-team/yorkie-js-sdk/blob/3a34270d67c07e3216ad765de9056c537cb77907/src/document/crdt/element_rht.ts#L172), so I will adjust it if it could cause any issues.

Reply:
fixed it to return nullable value [ac9080f](https://github.com/yorkie-team/yorkie-android-sdk/pull/84/commits/ac9080f54c5185448f83a87650659b76e236ff0b)